test/fuse_reduce.cpp

```diff
@@ -133,6 +134,53 @@ TEST_CASE(pointwise_reduce)
     EXPECT(p1 == p2);
 }
 
+TEST_CASE(scalar_multibroadcast)
+{
+    // Matches the find_pointwise_reduce matcher, but input x has a (scalar) shape
+    // incompatible with the multibroadcast instruction; therefore it
+    // creates a fused_reduce module but does not add a submodule for the
+    // multibroadcast instruction. 
+    migraphx::shape sdot{migraphx::shape::half_type, {80, 204, 204}};
+    migraphx::shape sdot_double{migraphx::shape::double_type, {80, 204, 204}};
+    migraphx::shape scalar{migraphx::shape::double_type, {1}, {0}};
+    migraphx::program p1;
+    {
+        auto* mm = p1.get_main_module();
+        auto x   = mm->add_parameter("x", scalar);
+        auto zap = add_pointwise(p1, "main:pointwise0", {x}, single_pointwise("sqrt"));
+        auto pow = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", sdot.lens()}}), zap);
+        auto bip    = mm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", {1, 2}}}), pow);
```

Comment:
Does the original model do a `reduce_sum` with a scalar input? If so, we should add an issue to track a perf improvement here as we can rewrite this as a multiply.

Reply:
No it doesn't.   It has 13 `reduce_sum` instructions, and they all have the same input dimensions as `@164` here:
```
@144 = @literal{ ... } -> half_type, {512}, {1}, target_id=0
@145 = @literal{ ... } -> half_type, {2048}, {1}, target_id=0
@146 = @literal{ ... } -> half_type, {512}, {1}, target_id=0
@147 = contiguous(arg1_1) -> half_type, {10, 204, 320}, {65280, 320, 1}, target_id=0
@148 = reshape[dims={2040, 320}](@147) -> half_type, {2040, 320}, {320, 1}, target_id=0
@149 = multibroadcast[out_lens={2040, 320},out_dyn_dims={}](@148) -> half_type, {2040, 320}, {320, 1}, target_id=0
@150 = multibroadcast[out_lens={320, 512},out_dyn_dims={}](@108) -> half_type, {320, 512}, {1, 320}, target_id=0
@151 = dot(@149,@150) -> half_type, {2040, 512}, {512, 1}, target_id=0
@152 = multibroadcast[out_lens={2040, 512},out_dyn_dims={}](@109) -> half_type, {2040, 512}, {0, 1}, target_id=0
@153 = multibroadcast[out_lens={2040, 512},out_dyn_dims={}](@151) -> half_type, {2040, 512}, {512, 1}, target_id=0
@154 = add(@152,@153) -> half_type, {2040, 512}, {512, 1}, target_id=0
@155 = contiguous(@154) -> half_type, {2040, 512}, {512, 1}, target_id=0
@156 = reshape[dims={10, 204, 512}](@155) -> half_type, {10, 204, 512}, {104448, 512, 1}, target_id=0
@157 = multibroadcast[out_lens={10, 204, 512},out_dyn_dims={}](@44) -> half_type, {10, 204, 512}, {0, 0, 0}, target_id=0
@158 = reduce_mean[axes={-1}](@156) -> half_type, {10, 204, 1}, {204, 1, 1}, target_id=0
@159 = multibroadcast[out_lens={10, 204, 512},out_dyn_dims={}](@158) -> half_type, {10, 204, 512}, {204, 1, 0}, target_id=0
@160 = sub(@156,@159) -> half_type, {10, 204, 512}, {104448, 512, 1}, target_id=0
@161 = multibroadcast[out_lens={10, 204, 512},out_dyn_dims={}](@43) -> half_type, {10, 204, 512}, {0, 0, 0}, target_id=0
@162 = div(@160,@161) -> half_type, {10, 204, 512}, {104448, 512, 1}, target_id=0
@163 = mul(@162,@162) -> half_type, {10, 204, 512}, {104448, 512, 1}, target_id=0
@164 = reduce_sum[axes={-1}](@163) -> half_type, {10, 204, 1}, {204, 1, 1}, target_id=0
@165 = multibroadcast[out_lens={10, 204, 512},out_dyn_dims={}](@164) -> half_type, {10, 204, 512}, {204, 1, 0}, target_id=0
@166 = add(@165,@157) -> half_type, {10, 204, 512}, {104448, 512, 1}, target_id=0
```